proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java

```diff
@@ -534,18 +530,29 @@ public static VelocityConfiguration read(Path path) throws IOException {
 
     String motd = config.getOrElse("motd", "<#09add3>A Velocity Server");
 
-    // MOTD Migration
+    // Old MOTD Migration
     if (configVersion < 2.6) {
-      if (!motd.startsWith("{")) {
-        final String migratedMotd = MiniMessage.miniMessage().serialize(
-                LegacyComponentSerializer.legacyAmpersand().deserialize(motd));
-
-        config.set("motd", migratedMotd);
-        motd = migratedMotd;
+      String migratedMotd = motd;
+
+      // JSON Format Migration
+      final String strippedMotd = migratedMotd.strip();
+      if (strippedMotd.startsWith("{\"") || strippedMotd.startsWith("[\"")) {
```

Comment:
imo it's only worth doing exactly the same check that was done when the component was retrieved originally (just the `{` check), as nothing else would've been displayed as JSON anyway